enterprise/backend/test/metabase_enterprise/advanced_permissions/common_test.clj

```diff
@@ -700,59 +700,70 @@
 (deftest upload-csv-test
   (mt/test-drivers (mt/normal-drivers-with-feature :uploads :schemas)
     (testing "Uploads should be blocked without data access"
-      (mt/with-empty-db
-        (let [table-name (mt/random-name)]
-          (let [conn-spec (sql-jdbc.conn/db->pooled-connection-spec (mt/db))]
-            (doseq [stmt ["CREATE SCHEMA IF NOT EXISTS \"not_public\";"
-                          (format "CREATE TABLE \"not_public\".\"%s\" (id INTEGER);" table-name)]]
-              (jdbc/execute! conn-spec stmt)))
-          (upload-test/sync-upload-test-table! :database (mt/db) :table-name table-name :schema-name "not_public")
-          (let [db-id    (u/the-id (mt/db))
-                table-id (t2/select-one-pk :model/Table :db_id db-id :name table-name)
-                upload-csv! (fn []
-                              (upload-test/upload-example-csv! {:grant-permission? false
-                                                                :schema-name       "not_public"
-                                                                :table-prefix      "uploaded_magic_"}))]
-            (doseq [[schema-perms can-upload?] {:all            true
-                                                :none           false
-                                                {table-id :all} false}]
-              (with-all-users-data-perms! {db-id {:data {:native :none, :schemas {"public"     :all
-                                                                                  "not_public" schema-perms}}}}
-                (if can-upload?
-                  (is (some? (upload-csv!)))
-                  (is (thrown-with-msg?
-                        clojure.lang.ExceptionInfo
-                        #"You don't have permissions to do that\."
-                        (upload-csv!)))))
-              (with-all-users-data-perms! {db-id {:data {:native :write, :schemas ["not_public"]}}}
-                (is (some? (upload-csv!)))))))))))
+      (mt/with-model-cleanup [:model/Table]
+        (let [table-name (mt/random-name)
+              schema-name (or (sql.tx/session-schema driver/*driver*) "public")
```

Comment:
Here's an example of how the change works.

Before we used the "not_public" schema and created tables in there, which wouldn't get cleaned up for redshift.

Now we use the result of sql.tx/session-schema for redshift, which is the `unique-session-schema` defined [here](https://github.com/metabase/metabase/blob/7dd097fd7728e879b1fe562d47c4252350aa2d47/modules/drivers/redshift/test/metabase/test/data/redshift.clj#L61)

"public" is used for postgres, mysql, and H2. The tests don't clean up the schema afterwards, only the tables.